cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/AzureImageService.java

```diff
@@ -36,96 +33,70 @@
     private AzureManagedImageService azureManagedImageService;
 
-    @Inject
-    private CustomVMImageNameProvider customVMImageNameProvider;
-
-    public AzureImage getCustomImageId(String resourceGroup, String fromVhdUri, AuthenticatedContext ac, boolean createIfNotFound, AzureClient client) {
-        String region = getRegion(ac);
-        String imageName = getImageName(region, fromVhdUri);
-        String imageId = getImageId(resourceGroup, client, imageName);
-        AzureManagedImageCreationCheckerContext checkerContext = new AzureManagedImageCreationCheckerContext(client, resourceGroup, imageName);
-
-        if (getCustomImage(resourceGroup, client, imageName).isPresent() || isRequested(imageId)) {
-            LOGGER.debug("Custom image found in '{}' resource group with name '{}'", resourceGroup, imageName);
-            azureManagedImageCreationPoller.startPolling(ac, checkerContext);
-            return new AzureImage(imageId, imageName, true);
-        } else {
-            LOGGER.debug("Custom image NOT found in '{}' resource group with name '{}', creating it now: {}", resourceGroup, imageName, createIfNotFound);
-            if (createIfNotFound) {
-                saveImage(ac, imageName, imageId);
-                Optional<VirtualMachineCustomImage> customImage;
-                try {
-                    customImage = Optional.of(client.createCustomImage(imageName, resourceGroup, fromVhdUri, region));
-                } catch (CloudException e) {
-                    customImage = handleImageCreationException(resourceGroup, ac, client, imageName, imageId, checkerContext, e);
-                }
-                return customImage
-                        .map(image -> createNewAzureImageAndNotify(ac, image))
-                        .orElseThrow(() -> new CloudConnectorException("Failed to create custom image."));
-            } else {
-                return null;
-            }
+    public Optional<AzureImage> findCustomImage(AzureImageDetails azureImageDetails, AzureClient client, AuthenticatedContext ac) {
+        if (findCustomImage(azureImageDetails, client).isEmpty() && !wasCreateRequested(azureImageDetails)) {
+            return Optional.empty();
         }
+
+        LOGGER.debug("Custom image found in '{}' resource group with name '{}'", azureImageDetails.getResourceGroup(), azureImageDetails.getImageName());
+        azureManagedImageCreationPoller.startPolling(ac, new AzureManagedImageCreationCheckerContext(azureImageDetails, client));
+        return Optional.of(new AzureImage(azureImageDetails.getImageId(), azureImageDetails.getImageName(), true));
     }
 
-    private AzureImage createNewAzureImageAndNotify(AuthenticatedContext ac, VirtualMachineCustomImage customImage) {
-        updateImageStatus(ac, customImage.name(), customImage.id(), CommonStatus.CREATED);
+    public AzureImage createCustomImage(AzureImageDetails azureImageDetails, String fromVhdUri, AzureClient client, AuthenticatedContext ac) {
```

Comment:
Generally, I don't like the image --> customImage renames, as it is an Azure Java SDK only term, the object type is "type": "Microsoft.Compute/images" and in Portal, it is called "Image" in Azure Portal, too.
Furthermore, the custom image term in Cloudbreak is used for a completely different concept. Pls. undo the renames. 
It would be best to rename all occurrences, regardless of this PR.

Reply:
You mean in AzureClient as well? Doing it.